### PR TITLE
[Windows] Added a test for _getFilesystemRepresentation

### PR DESCRIPTION
**Pull request by gmittert**
A small test to ensure that Windows FSRs don't regress

### CODE REVIEW

**Comment by compnerd** [comment]
@gmittert would you mind rebasing this change?  I think most of the issues are taken care of, but I think it would get good ensure that everything is merged before closing this out.

**Comment by gmittert** [comment]
@compnerd looks like you got all the those I found. The only thing left is the test that I added. I've rebased the change to just be the added test now.

**Comment by gmittert** [comment]
@swift-ci please test windows

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
Windows CI: https://dev.azure.com/compnerd/swift-build/_build/results?buildId=30730&view=results

File: Tests/Foundation/Tests/TestFileManager.swift [hunk 1]
```diff
@@ -1755,6 +1755,55 @@ VIDEOS=StopgapVideos
...
+            try "hello world".write(to: url, atomically: false, encoding: .utf8)
+            try fm.removeItem(atPath: path)
+            XCTAssertFalse(fm.fileExists(atPath: path))
+        }
+        let paths = [
+            // Absolute Paths
+            "\(tmpPath)\\testfile",       // C:/Users/...\testfile
+            "\(tmpPath)/testfile",        // C:/Users.../testfile
+            "\(tmpPath)/testfile",        // /Users/user/.../testfile
+            "\(tmpPath.first!):testfile", // C:testfile
```
Comment (lxbndr) [inline Tests/Foundation/Tests/TestFileManager.swift:1784]: Btw, this fails when runner has no Administrator privileges. CI agents are usually elevated, but manual run as user would report failure.

**Pull request merged**